### PR TITLE
Install gazelle from bazelbuild/bazel-gazelle instead of rules_go

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: downloads gazelle from its new home; it's being removed from `bazelbuild/rules_go`. It also removes @spiffxp's workaround from a few weeks ago.

**Special notes for your reviewer**: these should really be vendored (https://github.com/kubernetes/kubernetes/pull/57600), but this prevents us from running into issues in the meantime.

**Release note**:

```release-note
NONE
```

/approve no-issue
/assign @BenTheElder

### CODE REVIEW

File: hack/update-bazel.sh [hunk 1]
```diff
@@ -31,8 +31,8 @@ kube::util::go_install_from_commit \
     github.com/kubernetes/repo-infra/kazel \
     ae4e9a3906ace4ba657b7a09242610c6266e832c
 kube::util::go_install_from_commit \
-    github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle \
-    737df20c53499fd84b67f04c6ca9ccdee2e77089
+    github.com/bazelbuild/bazel-gazelle/cmd/gazelle \
+    31ce76e3acc34a22434d1a783bb9b3cae790d108  # 0.8.0
```
Comment (cblecker) [inline hack/update-bazel.sh]: as this is running a git checkout, couldn't we just put the tag here instead of the commit hash?

**Review by BenTheElder — APPROVED** [review]
/lgtm
/hold
(hold In case you want to switch to a tag)
I'm not convinced we shouldn't just use hashes everywhere though.

File: hack/update-bazel.sh (hunk 1 above)
Comment (ixdy) [reply hack/update-bazel.sh]: in theory, tags can be changed, whereas commits cannot. I'm not sure if it really matters here, but this feels a little more correct?

**Pull request merged**